### ISSUE TITLE
Include field name in validation errors

### ISSUE THREAD

**Issue by mathieufrh** [cloudcreativity/laravel-json-api#356]
This is more a question / feature request than an issue.

Is it possible to include the field name that failed the validation in the response ?
So that instead of:
```json
{
    "errors": [
        {
            "status": "422",
            "title": "Unprocessable Entity",
            "detail": "The requestor id field is required.",
            "source": {
                "pointer": "/data"
            }
        }
    ]
}
```

I would receive something like:
```json
{
    "errors": [
        {
            "status": "422",
            "title": "Unprocessable Entity",
            "detail": "The requestor id field is required.",
            "source": {
                "pointer": "/data"
            },
            "field": "requestor_id"
        }
    ]
}
```

Or maybe include it in the meta. Is this possible ? I really need to know the field name in order to process the errors in the front-end.

Thanks in advance.

**Comment by lindyhopchris** [comment]
**TLDR* use the JSON point in the error, and send a `null` value from the client for a required field that has no value.

So the JSON pointer is what you need to use as that tells you exactly where in the request JSON the problem came from.

There is only one instance where the pointer will be `/data` rather than the exact attribute or relationship. This is the `required` rule, and only occurs if your client does not send the required field. This is because according to the JSON pointer spec, the pointer can only point at a field that is present in the request.

To illustrate this, here are two requests that fail the `required` rule and the error response you'd get:

## Field Not Present

```json
{
  "data": {
    "type": "posts",
    "attributes": {
      "title": "Hello World"
    }
  }
}
```

```json
{
  "errors": [
    {
      "status": "422",
      "title": "Unprocessable Entity",
      "detail": "The content field is required.",
      "source": {
        "pointer": "/data"
      }
    }
  ]
}
```

I.e. the `/data` pointer indicates that the resource object in the `data` field is the problem. It cannot point to `/data/attributes/content` because that field does not exist in the request JSON.

## Field Empty

```json
{
  "data": {
    "type": "posts",
    "attributes": {
      "title": "Hello World",
      "content": null
    }
  }
}
```

```json
{
  "errors": [
    {
      "status": "422",
      "title": "Unprocessable Entity",
      "detail": "The content field is required.",
      "source": {
        "pointer": "/data/attributes/content"
      }
    }
  ]
}
```

I.e. the error can now point at `/data/attributes/content` because it is a field in the request JSON.

## Conclusion

You can solve this on the client side by sending `null` for any fields for which there are no values.

I think this answers your question, please close the issue if it does!

**Comment by mathieufrh** [comment]
Hey, thank you for the answer and thanks for updating the documentation.
This is what I was looking for but as you mention I was trying to get the name of a required field, that is not present in the JSON POST request. I will try to implement the "default to null" solution, this should not be too difficult.
I would prefer to receive a response with the name of the missing required fields even if they aren't part of the request but this behavior may not comply with the JSON API specifications.

Anyway, thank you !

**Comment by lindyhopchris** [comment]
No problem!
